src/libraries/Microsoft.PowerFx.Core/Public/FormulaSet.cs

```diff
@@ -0,0 +1,110 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.PowerFx.Core.Utils;
+
+namespace Microsoft.PowerFx.Core.Public
+{
+    /// <summary>
+    /// A set of name/formula paris, which creates and updates a topologically-sorted list
+    /// of name/formula pairs.
+    /// </summary>
+    public class FormulaSet
```

Comment:
Why public?

Reply:
This could be a very useful class in cases where you have multiple formulas and you want a topological sort for those formulas.

Reply:
Wouldn't we expect the RecalcEngine APIs to manage that, or is this more for other back ends eventually?